### PR TITLE
Rkhrabatyn - Added Order test 

### PR DESCRIPTION
**Pull request by romankhr**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 1]
```diff
@@ -29,8 +29,31 @@ public OrderPage setPhone(String phone) {
...
     public OrderPage setCity(String city) {
-        $x("//div[@class='form__row js-city']//input[@name='search']").setValue(city).click();
-        $x(String.format("(//li[contains(text(),'%s')])[1]", city)).click();
+        $x("//div[@class='form__row js-city']//input[@name='search']").setValue(city)
+                .click();
+        $x(String.format("//div[@class='form__row js-city']//li[1]", city)).click();
         return this;
     }
+
+    public boolean isDisplayedConfirmOrderButton() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: isConfirmOrderButtonDisplayed()

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 2]
```diff
@@ -4,21 +4,44 @@
...
 
     //This added for future tests
     private OrderPage openOrderPage;
     private BasketPage openBasketPage;
+    private String searchString;
 
     //This added for future tests
     @BeforeMethod
-    public void addProductToBasketOpenOrder() {
+    public void putProductToBasketOpenOrder() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: why was this changed?

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 3]
```diff
@@ -4,21 +4,44 @@
...
+        String surname = "Петренко";
+        String name = "Вася";
+        String city = "Львів";
+        String phone = "0997003330";
+        boolean isDisplayedConfirmOrderButton = openOrderPage
+                .setName(name)
+                .setSurname(surname)
+                .setCity(city)
+                .setPhone(phone)
+                .isDisplayedConfirmOrderButton();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: group these data into ShippingAddress class, add method setShippingAddress() to openOrderPage

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 4]
```diff
@@ -4,21 +4,44 @@
...
+    public void putProductToBasketOpenOrder() {
         openOrderPage = new HomePage()
                 .openHomePage()
                 .openProductByNumber(1)
                 .addProductToBasket()
                 .openOrderPage();
     }
+
+    @Test
+    public void insertBuyersCredentialsTest() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: rename

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java [hunk 5]
```diff
@@ -17,27 +18,31 @@ public OrderPage openOrderPage() {
...
+                .shouldBe(not(empty))
                 .val());
     }
 
     public int getOrderProductSum(int numberOfProduct) {
-        $x(String.format("(//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid'])[%d]", numberOfProduct)).click();
-        String ProductPrice = ($x(String.format("(//*[@class='cart-product__price'])[%d]", numberOfProduct))
-                .shouldBe(not(Condition.empty))
+        $x(format("(//*[@class='cart-counter__input ng-untouched ng-pristine ng-valid'])[%d]", numberOfProduct)).click();
+        var productPrice = ($x(format("(//*[@class='cart-product__price'])[%d]", numberOfProduct))
```
Comment (VladimirPetrat) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java]: productPriceText?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 6]
```diff
@@ -23,14 +13,35 @@ public String getPageTitle() {
...
+        return $x("//div[@class='form__row js-surname']/input").getValue();
+    }
+
+    public static String getName() {
+        return $x("//div[@class='form__row js-name']/input").getValue();
+    }
+
+    public static String getPhone() {
+        return $x("//input[@type='tel']").getValue()
+                .replaceAll(" ", "");
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:33]: new lines with two or more methods

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 7]
```diff
@@ -4,21 +4,45 @@
...
                 .openProductByNumber(1)
                 .addProductToBasket()
                 .openOrderPage();
     }
+
+    @Test
+    public void testShippingAddressInsertion() {
+        var shippingAddress = new ShippingAddress("Петренко", "Вася", "Львів", "0997003330");
+
+        var isDisplayedConfirmOrderButton = orderPage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: rename

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java (hunk 6 above)
Comment (romankhr) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:33]: Fixed

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 8]
```diff
@@ -1,37 +1,50 @@
...
-        $x("//div[@class='form__row js-city']//input[@name='search']").setValue(city).click();
-        $x(format("(//li[contains(text(),'%s')])[1]", city)).click();
+    public OrderPage setShippingAddress(ShippingAddress shippingAddress) {
+        $x("//div[@class='form__row js-surname']/input").setValue(shippingAddress.getSurname());
+        $x("//div[@class='form__row js-name']/input").setValue(shippingAddress.getName());
+        $x("//input[@type='tel']").setValue(shippingAddress.getPhone());
+        $x("//div[@class='form__row js-city']//input[@name='search']")
+                .setValue(shippingAddress.getCity())
+                .click();
+        $x("//div[@class='form__row js-city']//li[1]").click();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java:69]: you should have both setShippingAddress() and setSurname(), setName() etc methods and just those methods in setShippingAddress()

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 9]
```diff
@@ -21,4 +26,28 @@ public void addProductToBasketOpenOrder() {
                 .addProductToBasket()
                 .openOrderPage();
     }
+
+    @Test
+    public void testShippingAddressInsertion() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java [hunk 10]
```diff
@@ -21,4 +26,28 @@ public void addProductToBasketOpenOrder() {
...
+                .surname("Петренко")
+                .phone("0997003330")
+                .build();
+        var isConfirmOrderButtonDisplayed = orderPage
+                .setShippingAddress(shippingAddress)
+                .isConfirmOrderButtonDisplayed();
+
+        assertTrue(isConfirmOrderButtonDisplayed, "The Confirm Button should be displayed");
+        assertEquals(OrderPage.getSurname(), shippingAddress.getSurname(), "The surname should be "
+                + shippingAddress.getSurname());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/OrderTest.java]: Incorrect ...
same for all

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 11]
```diff
@@ -24,14 +14,58 @@ public String getHeaderText() {
...
+        return $x("//div[@class='form__row js-name']/input").getValue();
+    }
+
+    public static String getPhone() {
+        return $x("//input[@type='tel']")
+                .getValue()
+                .replaceAll(" ", "");
+    }
+
+    public static String getCity() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: no static methods in PO

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java [hunk 12]
```diff
@@ -24,14 +14,58 @@ public String getHeaderText() {
...
+
+    private OrderPage setPhoneNumber(String phoneNumber) {
+        $x("//input[@type='tel']").setValue(phoneNumber);
         return this;
     }
 
-    public OrderPage setCity(String city) {
-        $x("//div[@class='form__row js-city']//input[@name='search']").setValue(city).click();
-        $x(format("(//li[contains(text(),'%s')])[1]", city)).click();
+    private OrderPage setCity(String city) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java]: these methods should be public

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request merged**